### PR TITLE
[dbus] Enable default DBUS_SESSION_SOCKET_DIR when cross-compiling

### LINKED ISSUE #40031: [dbus] Build failure when cross-compiling
Editing the portfile to pass `-DDBUS_SESSION_SOCKET_DIR=/tmp` to CMake fixes the issue.

### Operating system

macOS (ARM64)

### Steps to reproduce the behavior

```Shell
./vcpkg install dbus:x64-osx-release
```

### Failure logs

```
CMake Error at CMakeLists.txt:574 (message):
  cannot autodetect session socket directory when crosscompiling, pass
  -DDBUS_SESSION_SOCKET_DIR=...


-- Configuring incomplete, errors occurred!
```

### PR DESCRIPTION
**Pull request by WangWeiLin-MV**
Fix #40031.

Allow setting default `DBUS_SESSION_SOCKET_DIR`, default `/tmp`.

### Checklist

- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md).
- [ ] ~SHA512s are updated for each updated download.~
- [ ] ~The "supports" clause reflects platforms that may be fixed by this new version.~
- [ ] ~Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.~
- [x] Any patches that are no longer applied are deleted from the port's directory.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is added to each modified port's versions file.

### Test

The port cross-compile installation tests pass with the following triplets:
* x64-osx (on arm64-osx)

### CODE REVIEW

File: ports/dbus/0006-cmake.tempdir.patch [hunk 1]
```diff
@@ -0,0 +1,13 @@
+diff --git a/CMakeLists.txt b/CMakeLists.txt
+index 6d232307fe1b2a3966e5b775eb8a6394ccb71f0f..74e50dbe3e03364ace1bb02bfa2f6af7afb979f9 100644
+--- a/CMakeLists.txt
++++ b/CMakeLists.txt
+@@ -569,7 +569,7 @@ endif()
+ #### Find socket directories
+ set(DBUS_SESSION_SOCKET_DIR "" CACHE STRING "Default directory for session socket")
+ if(UNIX)
+-    if (CMAKE_CROSSCOMPILING)
++    if (0)
```
Comment (dg0yt) [inline ports/dbus/0006-cmake.tempdir.patch:10]: Before this patch, the code required setting `DBUS_SESSION_SOCKET_DIR` when crosscompiling.
After this patch, `DBUS_SESSION_SOCKET_DIR` will be overridden for cross builds as for host builds. Which seems to be worse than before.

Comment (WangWeiLin-MV) [reply ports/dbus/0006-cmake.tempdir.patch:10]: It just removes the explicit requirement for cross-compiling to let it to the same default as the host build, no new default values ​​introduced.

Comment (davidebeatrici) [reply ports/dbus/0006-cmake.tempdir.patch:10]: While it's true that the behavior is going to be the same as with a native host build, I believe we should change it for the reason I explained at https://github.com/microsoft/vcpkg/issues/40031#issuecomment-2243958666.

Comment (dg0yt) [reply ports/dbus/0006-cmake.tempdir.patch:10]: I don't say it shouldn't be changed.
But I say it probably shouldn't be changed like this.

(There is no "explicit requirement for cross-compiling". There is a requirement to set the value for cross compiling. It might even be useful to use `if(1)` for vcpkg, and set the value via `OPTIONS`: The user could simply override it in the triplet when needed.)

Comment (davidebeatrici) [reply ports/dbus/0006-cmake.tempdir.patch:10]: Yeah, I agree. My comment was in reply to @WangWeiLin-MV.

Comment (WangWeiLin-MV) [reply ports/dbus/0006-cmake.tempdir.patch:10]: @dg0yt @davidebeatrici  If it shouldn't be changed like this, then the original native build also needs to be modified?

Comment (davidebeatrici) [reply ports/dbus/0006-cmake.tempdir.patch:10]: Correct.

File: ports/dbus/portfile.cmake [hunk 2]
```diff
@@ -37,6 +38,7 @@ vcpkg_cmake_configure(
         "-DCMAKE_INSTALL_SYSCONFDIR=${CURRENT_PACKAGES_DIR}/etc/${PORT}"
         "-DWITH_SYSTEMD_SYSTEMUNITDIR=lib/systemd/system"
         "-DWITH_SYSTEMD_USERUNITDIR=lib/systemd/user"
+        "-DDBUS_SESSION_SOCKET_DIR=/tmp"
```
Comment (BillyONeal) [inline ports/dbus/portfile.cmake:41]: "/tmp" is clearly not the correct path on Windows. This seems like something upstream might have Opinions about?

Comment (WangWeiLin-MV) [reply ports/dbus/portfile.cmake:41]: https://gitlab.freedesktop.org/dbus/dbus/-/blob/68637759b916a9d6a824a01318a55a76f7f08ec7/CMakeLists.txt#L609-620
https://gitlab.freedesktop.org/dbus/dbus/-/blob/68637759b916a9d6a824a01318a55a76f7f08ec7/CMakeLists.txt#L657-660

This value only useful on UNIX. And this value could still be changed through `VCPKG_KEEP_ENV_VARS` which can be include in vcpkg ABI hash.

**Comment by BillyONeal** [comment]
Let me rephrase my feedback, I agree I was not clear before. Upstream clearly know how to set this value to `/tmp`, it is, after all, a default for them. There is a reason that they do not use that default when crosscompiling, and it is not clear to me that `/tmp` is correct, even if that value causes the build to succeed. If we are making changes to this behavior, upstream needs to be notified and given a chance to respond.

I think we also need an explanation of why this is the correct value in a crosscompiling scenario, particularly given that on macOS, arm64 native stuff is often put into `/opt` rather than `/`, so it's possible for this change to produce something which builds, but which doesn't actually behave as expected for macOS customers.

@davidebeatrici You are asking us to apply this behavior change, but first, please attempt to contact the upstream maintainers following their instructions at https://www.freedesktop.org/wiki/Software/dbus/ . Post a link back here showing that they've had a chance to look at that. Ideally, they agree, and an upstream fix can be done which will "fix" vcpkg customers in addition to other customers on macOS. Or, they don't agree, and describe what they want done instead.

If you can explain more concretely why `/tmp` is the correct location, we may consider applying the change if and only if upstream has had at least 30 days notice to respond and are unresponsive.

**Comment by davidebeatrici** [comment]
Upstream recommends setting `/tmp`: https://gitlab.freedesktop.org/dbus/dbus/-/issues/551#note_2849725

File: ports/dbus/portfile.cmake (hunk 2 above)
Comment (davidebeatrici) [reply ports/dbus/portfile.cmake:41]: Upstream confirmed: https://gitlab.freedesktop.org/dbus/dbus/-/issues/551#note_2849725

**Comment by dg0yt** [comment]
> Upstream recommends setting `/tmp`

What I read is: Upstream recommends using meson and setting `session_socket_dir`.

**Comment by davidebeatrici** [comment]
The option is `DBUS_SESSION_SOCKET_DIR` in their CMake project, which is why setting it to `/tmp` solves the issue.

Using Meson is another argument.

**Comment by dg0yt** [comment]
Anyways, we already identified the option, and someone has to form a valid PR. This one isn't.
